x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/rules_list.tsx

```diff
@@ -995,24 +1005,39 @@ export const RulesList = ({
       )}
     </>
   );
-  // if initial load, show spinner
-  const getRulesList = () => {
-    if (noData && !rulesState.isLoading && !ruleTypesState.isLoading) {
-      return authorizedToCreateAnyRules ? (
-        <EmptyPrompt
-          showCreateRuleButton={showCreateRuleButton}
-          onCTAClicked={() => setRuleFlyoutVisibility(true)}
-        />
-      ) : (
-        noPermissionPrompt
-      );
-    }
 
+  const showPrompt = noData && !rulesState.isLoading && !ruleTypesState.isLoading;
```

Comment:
can we create a component outside of this file?

Reply:
done